src/data/list/indexes.lean

```diff
@@ -0,0 +1,168 @@
+import data.list.basic
+import data.list.defs
+import data.list.range
+import data.list.zip
+import logic.basic
+
+universes u v
+
+open function
+
+namespace list
+
+variables {α : Type u} {β : Type v}
+
+
+section foldr_with_index
+
+/-- Specification of `foldr_with_index_aux`. -/
+def foldr_with_index_aux_spec (f : ℕ → α → β → β) (start : ℕ) (b : β)
+  (as : list α) : β :=
+foldr (uncurry f) b $ zip (range' start as.length) as
```

Comment:
This isn't the specification, it's an alternative implementation. Maybe I'm confused by what's going on here.

Reply:
Well, one way to specify a function is to say that it behaves like another (less efficient) function. That's my strategy for all these functions because I don't want to redo all the `foldr` lemmas for `foldr_with_index`.